R/parents_of_children.R

```diff
@@ -0,0 +1,74 @@
+require(data.table)
+
+#' @name parents_of_children
+#' 
+#' @description Vectorized implementation of "parents_of_child"
+#' 
+#' @param child_location_id [int] Location ID to pull parent ID for
+#' @param hierarchy [data.table] Hierarchy. Must have columns location_id, path_to_top_parent, and level.
+#' @param parent_level [int] Level of the parent hierarchy to query.
+#' @export 
+parents_of_children <- function(  
+  child_loc_ids, 
+  hierarchy, 
+  parent_level
+){
+  return(sapply(child_loc_ids, function(x) parents_of_child(x, hierarchy, parent_level)))
+}
+
+#' @name parents_of_child
+#' 
+#' @description Given a location ID, returns its parent ID at a given level. 
+#' 
+#' @param child_location_id [int] Location ID to pull parent ID for
+#' @param hierarchy [data.table] Hierarchy. Must have columns location_id, path_to_top_parent, and level.
+#' @param parent_level [int] Level of the parent hierarchy to query.
+#' 
+#' @export
+#' @return 
+parents_of_child <- function(
+  child_location_id, 
+  hierarchy, 
+  parent_level
+) {  
+  validate_parents_of_children_inputs(child_location_id, hierarchy, parent_level)
+  
+  all_parents = hierarchy[location_id == child_location_id, path_to_top_parent]
+  all_parents = as.integer(unlist(strsplit(all_parents, ",")))
+  
+  for (parent in all_parents){
+    # Loop through all parents of this child. 
+    # If the parent you're checking exists on parent_level, then return. 
+    if (hierarchy[location_id == parent, level] == parent_level){
+      return(parent)
+    }
+  }
+  # Ideally, this never triggers, but if it does you at least have a good error message.
+  stop("Oops, something went wrong inside parents_of_children! Revisit inputs.")
```

Comment:
You should include the arguments in a good error message

Reply:
Sure; done.